src/TSTransformer/macros/callMacros.ts

```diff
@@ -21,9 +23,9 @@ export const CALL_MACROS: MacroList<CallMacro> = {
 		return luau.call(luau.globals.assert, args);
 	},
 
-	typeOf: (state, node, expression, args) => luau.call(luau.globals.typeof, args),
+	$typeof: (state, node, expression, args) => luau.call(luau.globals.typeof, args),
```

Comment:
I don't think is a positive change. Like `assert`, it's a macro which maps one-to-one to its Luau function. The `$` prefix should indicate that the emit is significantly different or in some other way related to compiler behaviour.